From 6d8e9332e69ffc05d1e633c9ce82692444f2c9cc Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Tue, 3 May 2016 22:55:09 -0700
Subject: [PATCH 0396/1239] USB: XHCI: Remove ThunderX-specific code

This code is unused since moving to ECAM-based enumeration,
remove it

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 drivers/usb/host/xhci-thunderx.c | 78 --------------------------------
 1 file changed, 78 deletions(-)
 delete mode 100755 drivers/usb/host/xhci-thunderx.c

diff --git a/drivers/usb/host/xhci-thunderx.c b/drivers/usb/host/xhci-thunderx.c
deleted file mode 100755
index 395eabd04e..0000000000
--- a/drivers/usb/host/xhci-thunderx.c
+++ /dev/null
@@ -1,78 +0,0 @@
-/**
- * (C) Copyright 2014, Cavium Inc.
- *
- * SPDX-License-Identifier:	GPL-2.0+
-**/
-
-#include <common.h>
-#include <usb.h>
-#include <asm-generic/errno.h>
-#include <linux/compat.h>
-
-#include "xhci.h"
-#include <cavium/atf.h>
-#include <cavium/thunderx.h>
-
-/* Declare global data pointer */
-DECLARE_GLOBAL_DATA_PTR;
-
-/**
- * Contains pointers to register base addresses
- * for the usb controller.
- */
-struct thunderx_xhci {
-	struct xhci_hccr *hcd;
-};
-
-#define USBHX_PF_BAR0(x)	(0x868000000000ull + (x) * (1ull << 36))
-#define USBHX_UCTL_CTL(x)	(0x868000100000ull + (x) * (1ull << 36))
-
-#define UCTL_CTL_CSCLK_EN	(1 << 4)
-#define UCTL_CTL_HCLK_EN	(1 << 30)
-
-#define UCTL_CTL_CLK_EN		(UCTL_CTL_CSCLK_EN | UCTL_CTL_HCLK_EN)
-
-
-static struct thunderx_xhci thunderx[CONFIG_USB_MAX_CONTROLLER_COUNT];
-
-#define XHCI_PER_NODE 2
-
-int xhci_hcd_init(int index, struct xhci_hccr **hccr, struct xhci_hcor **hcor)
-{
-	struct thunderx_xhci *ctx = &thunderx[index];
-	int node = index / XHCI_PER_NODE;
-	u64 uctl_ctl;
-
-	index %= XHCI_PER_NODE;
-
-	if (node >= atf_node_count()) {
-		*hccr = NULL;
-		*hcor = NULL;
-		return -ENODEV;
-	}
-
-	uctl_ctl = readq(CSR_PA(node, USBHX_UCTL_CTL(index)));
-
-	debug("ThunderX-xhci %d at node %d uctl_ctl: %lx\n", index, node,
-	      (unsigned long)uctl_ctl);
-
-	if ((uctl_ctl & UCTL_CTL_CLK_EN) != UCTL_CTL_CLK_EN)
-		return -ENODEV;
-
-	ctx->hcd = (struct xhci_hccr *)CSR_PA(node, USBHX_PF_BAR0(index));
-
-	*hccr = (ctx->hcd);
-	*hcor = (struct xhci_hcor *)
-			((uintptr_t) *hccr +
-			 HC_LENGTH(xhci_readl(&(*hccr)->cr_capbase)));
-
-	debug("ThunderX-xhci: init hccr %p and hcor %p hc_length %d\n",
-		*hccr, *hcor,
-		(uint32_t)HC_LENGTH(xhci_readl(&(*hccr)->cr_capbase)));
-
-	return 0;
-}
-
-void xhci_hcd_stop(int index)
-{
-}
-- 
2.29.0

